parquet-column/src/main/java/org/apache/parquet/filter2/predicate/ContainsRewriter.java

```diff
@@ -118,15 +118,18 @@ public FilterPredicate visit(And and) {
     } else if (and.getRight() instanceof Contains) {
       right = and.getRight();
     } else {
-      return and;
+      right = and.getRight();
     }
 
-    if (left instanceof Contains) {
-      if (!(right instanceof Contains)) {
-        throw new UnsupportedOperationException(
-            "Contains predicates cannot be composed with non-Contains predicates");
-      }
+    // If two Contains predicates refer to the same column, optimize by combining into a single predicate
```

Comment:
This fixes a bug with the way Contains predicates were composed -- previously it would throw an error if you tried to do something like `and(contains(binaryColumn("foo", ...), contains(binaryColumn("bar"), ...))` because the columns didn't match. however, that is still a valid predicate -- it's just that they should not be attempted to be combined into a single `Contains` predicate.